### PR TITLE
refactor: Remove static state

### PR DESCRIPTION
**Pull request by astubbs**
Removes static state and manipulation. Depends on DI module and transaction-docs.

- [x] related: #126

### CODE REVIEW

File: parallel-consumer-core/src/test-integration/java/io/confluent/parallelconsumer/integrationTests/CloseAndOpenOffsetTest.java [hunk 1]
```diff
@@ -88,14 +84,13 @@ void setup() {
...
     @EnumSource()
-    @ResourceLock(value = OffsetMapCodecManager.METADATA_DATA_SIZE_RESOURCE_LOCK, mode = READ)
     void offsetsOpenClose(OffsetEncoding encoding) {
         var skip = UniLists.of(OffsetEncoding.ByteArray, OffsetEncoding.ByteArrayCompressed);
         assumeFalse(skip.contains(encoding));
 
-        // todo remove - not even relevant to this test? smelly
-        OffsetMapCodecManager.forcedCodec = Optional.of(encoding);
-        OffsetSimultaneousEncoder.compressionForced = true;
+        // todo test removal - not even relevant to this test?
```
Comment (astubbs) [inline parallel-consumer-core/src/test-integration/java/io/confluent/parallelconsumer/integrationTests/CloseAndOpenOffsetTest.java:91]: resolve

File: parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/internal/PCModuleTestEnv.java [hunk 2]
```diff
@@ -86,4 +94,46 @@ public Clock clock() {
         return mutableClock;
     }
 
+
+    @Setter
+    private int maxMetadataSize = 4096;
```
Comment (astubbs) [inline parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/internal/PCModuleTestEnv.java:99]: link to constant?

File: parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/offsets/OffsetEncodingBackPressureUnitTest.java [hunk 3]
```diff
@@ -78,114 +66,110 @@ void backPressureShouldPreventTooManyMessagesBeingQueuedForProcessing() throws O
         List<WorkContainer<String, String>> toSucceed = workIfAvailable.stream().filter(x -> !blockedOffsets.contains(x.offset())).collect(Collectors.toList());
         toSucceed.forEach(wm::onSuccessResult);
 
-        try {
+//        try {
```
Comment (astubbs) [inline parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/offsets/OffsetEncodingBackPressureUnitTest.java:69]: remove

File: parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/state/WorkManagerTest.java [hunk 4]
```diff
@@ -59,7 +59,7 @@
 public class WorkManagerTest {
 
     public static final String INPUT_TOPIC = "input";
-    public static final String OUTPUT_TOPIC = "output";
+    // todo delete? public static final String OUTPUT_TOPIC = "output";
```
Comment (astubbs) [inline parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/state/WorkManagerTest.java:62]: delete

File: parallel-consumer-vertx/src/test-integration/java/io/confluent/parallelconsumer/vertx/integrationTests/VertxConcurrencyIT.java [hunk 5]
```diff
@@ -64,7 +63,7 @@
  */
 @Testcontainers
 @Slf4j
-@Isolated
+//@Isolated
```
Comment (astubbs) [inline parallel-consumer-vertx/src/test-integration/java/io/confluent/parallelconsumer/vertx/integrationTests/VertxConcurrencyIT.java:66]: delete

File: parallel-consumer-vertx/src/test/java/io/confluent/parallelconsumer/vertx/VertxTest.java [hunk 6]
```diff
@@ -43,7 +42,7 @@
 import static org.awaitility.Awaitility.await;
 import static pl.tlinkowski.unij.api.UniLists.of;
 
-@Isolated
+//@Isolated
```
Comment (astubbs) [inline parallel-consumer-vertx/src/test/java/io/confluent/parallelconsumer/vertx/VertxTest.java:45]: delete

**Review by astubbs — COMMENTED** [review]
just a few things to today up

**Review by astubbs — COMMENTED** [review]
lgtm

**Review by astubbs — COMMENTED** [review]
lgtm

File: parallel-consumer-core/src/test/java/io/confluent/csid/utils/LongPollingMockConsumer.java [hunk 7]
```diff
@@ -172,7 +179,29 @@ private ConsumerRebalanceListener getRebalanceListener() throws NoSuchFieldExcep
...
+
+    @Override
+    public synchronized void subscribe(final Collection<String> topics) {
+        super.subscribe(topics);
+        subscribeWithRebalanceAndAssignment(topics, getNumberOfPartitionsToWorkWith());
+    }
+
+    @Override
+    public synchronized void subscribe(final Pattern pattern, final ConsumerRebalanceListener listener) {
+        throw new IllegalStateException(); // cannot determine what a pattern would correlate to
```
Comment (astubbs) [inline parallel-consumer-core/src/test/java/io/confluent/csid/utils/LongPollingMockConsumer.java:196]: Move comment into exception message

File: parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/ParallelEoSStreamProcessorTestBase.java [hunk 8]
```diff
@@ -1,26 +1,34 @@
...
+        PCModule module = createModule(parallelConsumerOptions);
+        parallelConsumer = module == null ?
+                new ParallelEoSStreamProcessor<>(parallelConsumerOptions) :
+                new ParallelEoSStreamProcessor<>(parallelConsumerOptions, module);
         super.parentParallelConsumer = parallelConsumer;
         return parallelConsumer;
     }
 
+    protected PCModule<String, String> createModule(final ParallelConsumerOptions parallelConsumerOptions) {
+        return null;
```
Comment (astubbs) [inline parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/ParallelEoSStreamProcessorTestBase.java:31]: Resolve

**Review by astubbs — COMMENTED** [review]
Identified a couple of issues

File: parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/AbstractParallelEoSStreamProcessorTestBase.java [hunk 9]
```diff
@@ -197,22 +200,25 @@ protected void instantiateConsumerProducer() {
     protected void subscribeParallelConsumerAndMockConsumerTo(String topic) {
         List<String> of = of(topic);
         parentParallelConsumer.subscribe(of);
-        consumerSpy.subscribeWithRebalanceAndAssignment(of, 2);
-    }
 
-    protected void setupParallelConsumerInstance(ProcessingOrder order) {
```
Comment (astubbs) [inline parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/AbstractParallelEoSStreamProcessorTestBase.java:203]: resolve

**Review by astubbs — COMMENTED** [review]
lgtm, get tests passing

**Review by astubbs — COMMENTED** [review]
lgtm!

**Comment by eddyv** [comment]
Closing - Stale.

**Pull request closed without merging**